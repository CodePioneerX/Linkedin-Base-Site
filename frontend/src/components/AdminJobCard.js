import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { useNavigate } from 'react-router-dom';
import axios from "axios";
import { Container, Row, Col, Dropdown, DropdownButton } from 'react-bootstrap';
import { Link } from "react-router-dom";
import { faGripVertical } from "@fortawesome/free-solid-svg-icons";
import '../Assets/css/Admin.css';

const AdminJobCard = (props) => {

  const navigate = useNavigate();

  // Define state variables
  const [reportMessages, setReportMessages] = useState("");
  
  const userLogin = useSelector((state) => state.userLogin);
  const { userInfo } = userLogin;

  const updateToken = useSelector((state) => state.updateToken);
  const { loading, auth } = updateToken;

  const banHandler = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    }
    const response = await axios.put(`http://insightwearai.sytes.net:8000/api/users/ban/${author_id}`, {}, config)
    window.location.reload()
  }

  const dismissReportHandler = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    }
    const response = await axios.delete(`http://insightwearai.sytes.net:8000/api/jobs/report/dismiss/${id}`, config)
    window.location.reload()
  }

  const deleteJobHandler = async () => {
    const response = await axios.delete(`http://insightwearai.sytes.net:8000/api/job/delete/${job_id}`)
    window.location.reload()
  }

  const viewAuthor = () => {
    navigate('/profileScreen', { state: {data: author_id} })
  }

  const viewSender = () => {
    navigate('/profileScreen', { state: {data: sender_id} })
  }

  const viewJob = () => {
    navigate('/job', { state: {job_id: job_id} })
  }

  const { id, sender_id, sender_name, sender_email, job_id, job_title, job_company, job_location, message, author_id, author_name } = props.report

  return (
    <>   
      <Container className="adminCardContainer">
        <div className="adminCardBody">
          <Row>
            <Col xs={8} md={10}>
              <Row className="adminCardRow">
                <h4>Job Author: {author_name}</h4>
              </Row>
              <Row className="adminCardRow">
                <h4>{job_title} at {job_company} in {job_location}</h4>
              </Row>
              <Row className="adminCardRow">
                  <p><strong>{sender_name}</strong> ({sender_email}): "{message}"</p>
              </Row>
              <Row className="adminCardRow">
                <hr style={{width: "100%"}}/>
              </Row>
            </Col>
            <Col xs={4} md={2} style={{display:'flex', justifyContent:'right'}}>
              <Row>
                <DropdownButton id="dropdownButton" title=''>
                  <Dropdown.Item as="button" onClick={viewAuthor}>View Author</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={viewSender}>View Sender</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={viewJob}>View Job</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={dismissReportHandler}>Dismiss Report</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={deleteJobHandler}>Delete Job</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={banHandler}>Ban User</Dropdown.Item>
                </DropdownButton>
              </Row>
            </Col>
          </Row>
        </div>
      </Container>
    </>
  );
};

export default AdminJobCard;
