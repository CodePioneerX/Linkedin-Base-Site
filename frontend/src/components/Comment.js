// Importing necessary packages from 'mdb-react-ui-kit' and 'react'
import React from 'react';
import {
  MDBCard,
  MDBCardBody,
  MDBCardImage,
  MDBCol,
  MDBContainer,
  MDBIcon,
  MDBInput,
  MDBRow,
} from "mdb-react-ui-kit";

// Functional component Comment with props 'author', 'content', 'createdAt'
const Comment = ({ author, content, createdAt }) => {
  return (
    // Using MDBContainer and MDBRow to create a container and row to hold the comment
    <MDBContainer>
        <MDBRow> 
    // Using MDBCard from MDBReact to create a card for the comment
        <MDBCard className="mb-4">
                <MDBCardBody>
                  // Adding a prompt for users to add their note
                  <p>Type your note, and hit enter to add it</p>
                  
                  // Dividing the card into two parts - left and right
                  <div className="d-flex justify-content-between">
    
                    // Left part with the author's image and name
                    <div className="d-flex flex-row align-items-center">
                      <MDBCardImage
                        src="https://mdbcdn.b-cdn.net/img/Photos/Avatars/img%20(4).webp"
                        alt="avatar"
                        width="25"
                        height="25"
                      />
                      <p className="small mb-0 ms-2">Martha</p>
                    </div>
    
                     // Right part with upvote count and an upvote icon
                    <div className="d-flex flex-row align-items-center">
                      <p className="small text-muted mb-0">Upvote?</p>
                      <MDBIcon
                        far
                        icon="thumbs-up mx-2 fa-xs text-black"
                        style={{ marginTop: "-0.16rem" }}
                      />
                      <p className="small text-muted mb-0">3</p>
                    </div>
                  </div>
                </MDBCardBody>
              </MDBCard>
        </MDBRow>
    </MDBContainer>
  );
};

export default Comment;
