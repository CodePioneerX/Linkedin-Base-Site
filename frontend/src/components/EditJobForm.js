// Import necessary components and functions
import Button from 'react-bootstrap/Button';
import React, { useState, useEffect } from 'react'
import { useDispatch, useSelector } from "react-redux";
import '../Assets/css/Login.css';
import { Form, Label, Input, Row, Col } from 'reactstrap';
import { delete_job, update_job } from '../actions/jobActions';
import { salary_types, employment_terms, job_types } from '../constants/jobConstants';

// Define the EditJobForm component with the job object as a parameter
export const EditJobForm = (job) => {
    
    // Define state variables to store the job details
    const [author, setAuthor] = useState(job.job.author)
    const [email, setEmail] = useState(job.job.author)
    const [title, setTitle] = useState(job.job.title)
    const [description, setDescription] = useState(job.job.description)
    const [remote, setRemote] = useState(job.job.remote)
    const [status, setStatus] = useState(job.job.status)
    const [company, setCompany] = useState(job.job.company)
    const [location, setLocation] = useState(job.job.location)
    const [image, setImage] = useState(job.job.image)
    const [deadline, setDeadline] = useState(job.job.deadline)
    const [requiredDocs, setRequiredDocs] = useState(job.job.required_docs)
    const [salary, setSalary] = useState(job.job.salary)
    const [salaryType, setSalaryType] = useState(job.job.salary_type)
    const [listingType, setListingType] = useState(job.job.listing_type)
    const [link, setLink] = useState(job.job.link)
    const [employmentTerm, setEmploymentTerm] = useState(job.job.employment_term)
    const [jobType, setJobType] = useState(job.job.job_type)

    // Retrieve the user's login information from the Redux store
    const userLogin = useSelector((state) => state.userLogin)
    const { userInfo } = userLogin;
    
    // Define the dispatch function to update or delete the job
    const dispatch = useDispatch();
    
    // Define a function to handle the submission of the job form
    const submitHandler = (e)=>{
        e.preventDefault()
        // Dispatch an action to update the job details
        dispatch(update_job(job.job.id, author, title, description, remote, status, company, image, location, deadline, requiredDocs, salary, salaryType, listingType, link, employmentTerm, jobType)).then(
            (res) => {
                if (res.success) {
                    window.location.reload(false)            
                } else {
                    alert(res.message)
                }
            }
        )
    }
    
    // Define a function to handle the cancellation of the job form
    const cancelHandler = (e) => {
        e.preventDefault()
        
         // Reload the page
        window.location.reload(false)
    }
    
    // Define a function to handle the deletion of the job
    const deleteHandler = (e) => {
        e.preventDefault()
        // Alert the user to confirm the deletion of the job
        alert('Are you sure you want to delete this job?')
        // Dispatch an action to delete the job
        dispatch(delete_job(job.job.id))
        // Reload the page
        window.location.reload(false)
    }

    // On page load, converts deadline to proper format
    // and sets the min value of the deadline input to current date
    useEffect(() => {
        var deadline_date = deadline.slice(0, 10)
        setDeadline(deadline_date)

        const date = new Date()
    
        let day = date.getDate();
        let month = date.getMonth() + 1;
        (month < 10) && (month = `0${month}`);
        let year = date.getFullYear();
        
        let today = `${year}-${month}-${day}`;
        
        document.getElementById("deadline").setAttribute("min", today);
    }, []);

    // updates requiredDocs based on checkbox selection
    function updateRequiredDocs(document) {
        const newDocs = requiredDocs.map(doc => {
          if (doc.type === document) {
            return {
                ...doc, required: !(doc.required)
              };
          }
          else {
            return doc;
          }
        })
        setRequiredDocs(newDocs)
      }
    
    // Render the form for editing the job details
    return <div>
        <h2>Edit Your Job Listing</h2>
        <Form onSubmit={submitHandler}>
            <Row className='mb-4'>
                <Col>
                    <Label className='labelE' for='contact-name'>Contact Name</Label>
                    <Input name='contact-name' disabled id='contact-name' value={userLogin.userInfo.username} onChange={(e)=> setAuthor(e.target.value)} />
                </Col>
                <Col>
                    <Label className='labelE' for='contact-email'>Contact Email</Label>
                    <Input name='contact-email' disabled id='contact-email' value={userLogin.userInfo.username} onChange={(e)=> setEmail(e.target.value)}/>
                </Col>
            </Row>
            <Row className='mb-4'>
                <Col>
                    <Label className='labelE' for='job-title'>Job Title</Label>
                    <Input name='job-title' id='job-title' required={true} value={title} onChange={(e)=> setTitle(e.target.value)} />
                </Col>
                <Col>
                    <Label className='labelE' for='company'>Company</Label>
                    <Input name='company' id='company' required={true} value={company} onChange={(e)=> setCompany(e.target.value)}/>
                </Col>
            </Row>
            <Row className='mb-4'>
                <Col>
                    <Label className='labelE' for='description'>Description</Label>
                    <Input name='description' type='textarea' rows='4' id='description' required={true} value={description} onChange={(e)=> setDescription(e.target.value)}/>
                </Col>
            </Row>
            <Row className='mb-4'>
                <Col>
                    <Label className='labelE' for='location'>Location</Label>
                    <Input name='location' id='location' required={true} value={location} onChange={(e)=> setLocation(e.target.value)}/>
                </Col>
                <Col>
                    <Label className='labelE' for='job-type'>Job Type (Full/Part-Time, etc.)</Label>
                    <Input name='job-type' id='job-type' type='select' required={true} value={jobType} onChange={(e)=> setJobType(e.target.value)}>
                            {job_types.map(type => (
                                <option key={type.value} value={type.value}>{type.name}</option>
                            ))}
                    </Input>
                </Col>
                <Col>
                    <Label className='labelE' for='employment-term'>Employment Term</Label>
                    <Input name='employment-term' id='employment-term' type='select' required={true} value={employmentTerm} onChange={(e)=> setEmploymentTerm(e.target.value)}>
                        {employment_terms.map(type => (
                            <option key={type.value} value={type.value}>{type.name}</option>
                        ))}
                    </Input>
                </Col>
            </Row>
            <Row className='mb-4'>
                <Col>
                    <Label className='labelE' for='salary'>Salary</Label>
                    <Input name='salary' id='salary' type='number' label='Salary' required={true} value={salary} onChange={(e)=> setSalary(e.target.value)}/>    
                </Col>
                <Col>
                    <Label className='labelE' for='salary-type'>Salary Type</Label>
                    <Input name='salary-type' id='salary-type' type='select' required={true} value={salaryType} onChange={(e) => setSalaryType(e.target.value)}>
                        {salary_types.map(type => (
                            <option key={type.value} value={type.value}>{type.name}</option>
                        ))}
                    </Input>
                </Col>
                <Col>
                    <Label className='labelE' for='deadline'>Application Deadline</Label>
                    <Input name='deadline' id='deadline' type='date' required={true} value={deadline} onChange={(e)=> setDeadline(e.target.value)}/>
                </Col>
            </Row>
            <Row className='mb-4'>
                <Col style={{paddingLeft: "40px"}}>
                    <Input name='remote' id='remote' className='form-checkbox-input' type="checkbox" checked={remote} onChange={(e)=> setRemote(!remote)}/>
                    <Label className='labelE' for='remote'>Remote?</Label>
                </Col>
                <Col style={{paddingLeft: "40px"}}>
                    <Input name='status' id='status' className='form-checkbox-input' type='checkbox' checked={status} onChange={(e)=> setStatus(!status)}/>
                    <Label className='labelE' for='status'>Active?</Label>
                </Col>
                <Col style={{paddingLeft: "40px"}}>
                    <Input name='listing-type' id='listing-type' className='form-checkbox-input' type='checkbox' checked={(listingType == 'INTERNAL') ? false : true} onChange={(e) => {
                        (listingType == 'INTERNAL') ? 
                        setListingType('EXTERNAL')
                        :
                        setListingType('INTERNAL')
                        }}/>
                    <Label className='labelE' for='listing-type'>External?</Label>
                </Col>
            </Row>
            {(listingType == 'EXTERNAL') &&  
                <Row className='mb-4'>
                    <Col>
                        <Label className='labelE' for='link'>Link to Application</Label>
                        <Input name='link' id='link' required={true} value={link} onChange={(e) => setLink(e.target.value)}/>
                    </Col>
            </Row>}
            <Row className='mb-4'>
                <Col>
                    <Label className='labelE' for='listing-image'>Listing Image</Label>
                    <Input name='listing-image' id='listing-image' type="file" onChange={(e)=> setImage(e.target.files[0])}/>
                </Col>
            </Row>
            {requiredDocs.map((doc, index) => (
                <Row key={index}>
                    <Col xs='2'>
                        <Input name={doc.type} id={doc.type} className='form-doc-input' type='checkbox' checked={doc.required} onChange={() => updateRequiredDocs(doc.type)} />
                    </Col>
                    <Col xs='10'>
                        <Label className='labelE' style={{position:'relative', left:'-100px'}} for={doc.type}>{`${doc.type}`} Required</Label>
                    </Col>
                </Row>
            ))}
            <Row className='mb-4'></Row>
            <Row className='editButtonContainer'>
                <Col xs={12} md={4}>
                    <Button type = 'submit' className='editSaveButton'> Save </Button>
                </Col>
                <Col xs={12} md={4}>
                    <Button className='editCancelButton' onClick={cancelHandler}>Cancel</Button>
                </Col>
                <Col xs={12} md={4}>
                    <Button className='editDeleteButton' onClick={deleteHandler}>Delete</Button>
                </Col>
            </Row>
        </Form>
    </div>
} 
