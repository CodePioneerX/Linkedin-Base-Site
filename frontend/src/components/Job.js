import React, { Component } from 'react'
import {Container, Row, Col, Card } from 'react-bootstrap';
import { Link, useNavigate } from 'react-router-dom';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import axios from 'axios'
import Comment from './Comment';
import Message from './Message';
import {
    MDBCard,
    MDBCardBody,
    MDBCardImage,
    MDBCol,
    MDBContainer,
    MDBIcon,
    MDBInput,
    MDBRow,
    MDBPagination, 
    MDBPaginationItem, 
    MDBPaginationLink
  } from "mdb-react-ui-kit";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPenToSquare } from '@fortawesome/free-regular-svg-icons';

import { salary_types, employment_terms, job_types } from '../constants/jobConstants';

export default class Job extends Component {
    state = {
        job: [],
        error: ''
    }

    componentDidMount() {
        try {
            axios.get(`http://localhost:8000/api/job/${this.props.job_id}`)
                .then(res => this.setState({ job: res.data[0], required_docs: res.data[1], ready: true }))
                .catch(() => this.setState({error: true}))
       } catch(error) {
        console.log(error)
       }
    }

    handleClick = job => () => {
        let required_docs = this.state.required_docs

        let job_ = {...this.state.job, required_docs}
        
        this.props.setjob(job_)
        this.props.edit()
    }

  render() {
      const Comment = ({ author, content, createdAt }) => {
          return (
            <div style={{ marginBottom: 20 }}>
            <p>Author: {author}</p>
            <p>Content: {content}</p>
            <p>Created at: {createdAt}</p>
          </div>
        );
    };

    if (this.state.error) return (
    <>
        
        <Message variant='danger'>There was an error while loading job data. Please try again later.</Message>
    </>
    ) 
    if (this.state.ready) return (
      <>
        <Row style={{marginBottom: "25px", width: "100%"}}>
            <Container style={{paddingBottom:"80px"}}>
            <Col xs={12} style={{ borderRadius: "20px", boxShadow: "0 0 5px rgba(0, 0, 0, 0.3)", padding: "25px", backgroundColor: "white", border: "none", marginBottom: "-40px" }}>
            <div style={{borderBottom: "1px solid #d3d3d3",marginBottom:"10px"}}></div>
            <Row className='mb-4'>
                <Col xs={8} md={10}>
                <span>
                    <h2>{this.state.job.title} at {this.state.job.company}</h2>
                </span>
                </Col>
                <Col xs={4} md={2} style={{display:'flex', alignItems: 'center', justifyContent: 'end'}}>
                {this.props.author == this.state.job.author ? <button onClick={this.handleClick(this.state.job)} style={{ backgroundColor: "#3D13C6", color: "white", borderRadius: "25px", padding: "5px 10px", border: "none" }}><FontAwesomeIcon icon={faPenToSquare} style={{ color: "white"}}/> </button> : <></>}
                </Col>
            </Row>
            <Container className='darker'>
                {this.state.job.image ? <img src={'http://localhost:8000'+this.state.job.image} alt={this.state.job.title} style={{ borderRadius: "50%", width: "auto", height: "55px", marginRight: "10px" }} /> : <></>}
                <Row>
                <Col xs={12}>
                    <h4>Description: </h4>
                    <p style={{whiteSpace: 'pre-wrap'}}>{this.state.job.description}</p>
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Location: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p >{this.state.job.location}</p>
                    {this.state.job.remote ? <p>Remote work possible.</p> : <p>Must be willing to work in person.</p>} 
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Salary: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p>${this.state.job.salary} {salary_types.map(type => ((type.value == this.state.job.salary_type) && <span key={type.value}>{type.name}</span>))}</p>
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Position Type: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p>{job_types.map(type => ((type.value == this.state.job.job_type) && <span key={type.value}>{type.name}</span>))}</p>
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Employment Term: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p>{employment_terms.map(type => ((type.value == this.state.job.employment_term) && <span key={type.value}>{type.name}</span>))}</p>
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Position Type: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p>{job_types.map(type => ((type.value == this.state.job.job_type) && <span key={type.value}>{type.name}</span>))}</p>
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Posted: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p>{this.state.job.created_at.slice(0, 10)}</p>
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Application Deadline: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p>{this.state.job.deadline.slice(0, 10)}</p>
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Status: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p>{this.state.job.status ? <>Applications Open</> : <>Applications Closed</>}</p>
                </Col>
                </Row>
                <Row className='mb-2'>
                <Col sm={12} md={6} xl={4}>
                    <h4>Application Type: </h4>
                </Col>
                <Col sm={12} md={6} xl={8}>
                    <p>{(this.state.job.listing_type == 'INTERNAL') ? <>Internal</> : <a href={this.state.job.link}>External</a>}</p>
                </Col>
                </Row>
                <Row className='mb-1'>
                <Col sm={12} md={6} xl={4}>
                    {this.state.required_docs.find(doc => doc.required === true) && <h4>Required Documents: </h4>}
                </Col>
                <Col sm={12} md={6} xl={8}>
                    {this.state.required_docs && 
                    <>
                        <ul key={this.state.job.id} style={{paddingLeft:'20px'}}>
                        {this.state.required_docs.map(doc => (
                        <React.Fragment key={doc.type}>
                            {doc.required && 
                            <><li>{doc.type}</li><br/></>}
                        </React.Fragment>
                        ))}
                        </ul>
                    </>
                    }
                </Col>
                </Row>
            {this.state.job.comments && this.state.job.comments.length > 0 && <h4>Comments</h4>}
            {this.state.job.comments && this.state.job.comments.length > 0 && this.state.job.comments.map(comment => (
                
                <Row>
                        <MDBContainer >
                        <MDBRow> 
                        <MDBCard className="mb-4 restrict-height">
                                <MDBCardBody className='common-width '>
                                    <Row> 
                                

                                <div className="d-flex justify-content-between common-width">
                                    <Col>
                                        <div className="d-flex flex-row align-items-center">
                                        <MDBCardImage
                                            className="rounded-circle"
                                            src='http://localhost:8000/images/default.png'
                                            alt="avatar"
                                        />
                                        <p className="tiny">{comment.author}</p>
                                        </div>
                                    </Col>
                                    <Col><p className="small">{comment.content}</p></Col>
                                    <Col>
                                        <div className="d-flex flex-row align-items-center">
                                        <p className="tiny text-muted ">Likes ( </p>
                                        <MDBIcon
                                            far
                                            icon="thumbs-up fa-xs text-black"
                                            
                                        />
                                        <p className="small text-muted "> 3)</p>
                                        </div>
                                    </Col>
                                </div>
                                </Row>
                                </MDBCardBody>
                            </MDBCard>
                        </MDBRow>
                    </MDBContainer>
                </Row>
            ))}
            <Row className='mb-4'>
                <Form.Control className='padd pad_down' type="text" placeholder="Comment..." />
            </Row>
            </Container>

            <Row>
                <Col style={{display: 'flex', justifyContent: 'center'}}>
                {this.state.job.listing_type == 'INTERNAL' ? 
                    <Button className='btns' variant="primary">Apply</Button> : 
                    <>
                    <Link to={this.state.job.link}>
                        <Button className='btns' variant="primary">Apply Externally</Button>
                    </Link>
                    </>
                }
                </Col>
                <Col style={{display: 'flex', justifyContent: 'center'}}>
                    <Button className='btns' variant="primary">Like</Button>
                </Col>
                <Col style={{display: 'flex', justifyContent: 'center'}}>
                    <Button className='btns' variant="secondary">Comment</Button>
                </Col>
            </Row>
            </Col>
            </Container>
        </Row>
    </>
    )
  }
}
