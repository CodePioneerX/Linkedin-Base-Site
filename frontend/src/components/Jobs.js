import React, { Component } from 'react'
import {Container, Row, Col, Card } from 'react-bootstrap';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import axios from 'axios'
import Comment from './Comment';
import {
    MDBCard,
    MDBCardBody,
    MDBCardImage,
    MDBCol,
    MDBContainer,
    MDBIcon,
    MDBInput,
    MDBRow,
    MDBPagination, 
    MDBPaginationItem, 
    MDBPaginationLink
  } from "mdb-react-ui-kit";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPenToSquare } from '@fortawesome/free-regular-svg-icons';

export default class Jobs extends Component {
  state = {
    jobs: []
  }

  componentDidMount() {
    axios.get('http://localhost:8000/api/jobs/')
      .then(res => this.setState({ jobs: res.data }))
  }

  handleClick = job => () => {
    this.props.setjob(job)
    this.props.edit()
  }

  render() {
    const Comment = ({ author, content, createdAt }) => {
        return (
          <div style={{ marginBottom: 20 }}>
            <p>Author: {author}</p>
            <p>Content: {content}</p>
            <p>Created at: {createdAt}</p>
          </div>
        );
      };

    return (
      <>
      <Container className="justify-content-md-center padd">
        <Row>
            <h1>Latest Jobs</h1>
        </Row>
        <Row style={{display: "flex", flexDirection: "column", alignItems: "center", marginBottom:"100px"}}>
            {this.state.jobs.map(job => (
              <Row key={job.id} style={{marginBottom: "25px", width: "75%"}}>
              <Container style={{padding: "35px 0 0 35px", paddingBottom: "0" }}>
                <Col xs={10} style={{ borderRadius: "20px", boxShadow: "0 0 5px rgba(0, 0, 0, 0.3)", padding: "25px", backgroundColor: "white", border: "none", marginBottom: "-40px" }}>
                <div style={{borderBottom: "1px solid #d3d3d3",marginBottom:"10px"}}></div>
                <Row>
                  <Col xs={8} md={10}>
                    <h3>{job.title} at {job.company}</h3>
                  </Col>
                  <Col xs={4} md={2}>
                    {this.props.author == job.author ? <button onClick={this.handleClick(job)} style={{ backgroundColor: "#3D13C6", color: "white", borderRadius: "25px", padding: "5px 10px", border: "none" }}><FontAwesomeIcon icon={faPenToSquare} style={{ color: "white"}}/> </button> : <></>}
                  </Col>
                </Row>
                <Container className='darker'>
                  <p>{job.content}</p>
                  {/* <img className='img_box' src={('http://localhost:8000/images/'+job.image.split('images/')[1])} /> */}
                  {job.image ? <img src={'http://localhost:8000/'+job.image} alt={job.title} style={{ borderRadius: "50%", width: "auto", height: "55px", marginRight: "10px" }} /> : <></>}
                  <p>Description: {job.description}</p>
                  <p>Location: {job.location}</p>
                  <p>Salary: {job.salary}$/hour</p>
                  <p>Position type: {job.job_type}</p>
                  {job.remote ? <p>Remote work possible.</p> : <p>Must be willing to work in person.</p>} 
                  <p>Recruiter: {job.author}</p>
                  <p>Likes: {job.likes}</p>
                  <p>Posted at: {job.created_at}</p>
                  <br/>

                {job.comments.length > 0 && <h4>Comments</h4>}
                {job.comments.length > 0 && job.comments.map(comment => (
                    
                    <Row>
                            <MDBContainer >
                            <MDBRow> 
                            <MDBCard className="mb-4 restrict-height">
                                    <MDBCardBody className='common-width '>
                                      <Row> 
                                    

                                    <div className="d-flex justify-content-between common-width">
                                        <Col>
                                            <div className="d-flex flex-row align-items-center">
                                            <MDBCardImage
                                                className="rounded-circle"
                                                src='http://localhost:8000/images/default.png'
                                                alt="avatar"
                                            />
                                            <p className="tiny">{comment.author}</p>
                                            </div>
                                        </Col>
                                        <Col><p className="small">{comment.content}</p></Col>
                                        <Col>
                                          <div className="d-flex flex-row align-items-center">
                                          <p className="tiny text-muted ">Likes ( </p>
                                          <MDBIcon
                                              far
                                              icon="thumbs-up fa-xs text-black"
                                              
                                          />
                                          <p className="small text-muted "> 3)</p>
                                          </div>
                                        </Col>
                                    </div>
                                    </Row>
                                    </MDBCardBody>
                                </MDBCard>
                            </MDBRow>
                        </MDBContainer>
                    </Row>
                ))}
                  
                  <Form.Control className='padd pad_down limit_width' type="text" placeholder="Comment..."  rows={4} />
                  <br/>
                </Container>

                <Row>
                  <Col>
                    <Button className='btns' variant="primary">Apply</Button>
                  </Col>
                  <Col>
                    <Button className='btns' variant="primary">Like</Button>
                  </Col>
                  <Col>
                    <Button className='btns' variant="secondary">Comment</Button>
                  </Col>
                  
                </Row>
                </Col>
                </Container>
              </Row>
            ))}

        </Row>
        <Row>
        <nav aria-label='Page navigation example'>
          <MDBPagination className='mb-0'>
            <MDBPaginationItem>
              <MDBPaginationLink href='#'>Previous</MDBPaginationLink>
            </MDBPaginationItem>
            <MDBPaginationItem>
              <MDBPaginationLink href='#'>1</MDBPaginationLink>
            </MDBPaginationItem>
            <MDBPaginationItem>
              <MDBPaginationLink href='#'>2</MDBPaginationLink>
            </MDBPaginationItem>
            <MDBPaginationItem>
              <MDBPaginationLink href='#'>3</MDBPaginationLink>
            </MDBPaginationItem>
            <MDBPaginationItem>
              <MDBPaginationLink href='#'>Next</MDBPaginationLink>
            </MDBPaginationItem>
          </MDBPagination>
        </nav>
        </Row>
      </Container>
    </>
    )
  }
}
