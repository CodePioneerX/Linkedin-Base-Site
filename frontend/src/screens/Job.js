import React, { Component } from 'react'
import {Container, Row, Col} from 'react-bootstrap';
import { useDispatch, useSelector } from "react-redux";
import Jobs from '../components/Jobs';
import Button from 'react-bootstrap/Button';
import axios from 'axios'
import Alert from 'react-bootstrap/Alert';

function Job() {
  const userLogin = useSelector((state) => state.userLogin);
  const { userInfo } = userLogin;


  return (
      <Container className="justify-content-md-center padd">
        {userInfo ? (
          <></>
        ) : (        
        <Row>
          
          <Alert  className='alertLogin' key='primary' variant='primary'>
              <h5>You are not signed in! Please <a href="/login">Sign in</a> or <a href="/register">Register</a>!</h5>
          </Alert>

        </Row>)}
        
        <Jobs/>
      </Container>
  )

}
export default Job;