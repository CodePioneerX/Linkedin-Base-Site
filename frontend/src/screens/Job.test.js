import React from 'react';
import { shallow, configure} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'; // import the adapter
import { Provider } from 'react-redux';
import { useSelector } from 'react-redux';
import Job from './Job';
import { EditJobForm } from '../components/EditJobForm';
import Jobs from '../components/Jobs';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

// configure Enzyme to use the adapter
configure({ adapter: new Adapter() });


describe('Job Component', () => {
  let wrapper;
  const userInfo = { email: 'user@example.com' };
  const setJobEditorMock = jest.fn();
  const setJobMock = jest.fn();

  beforeEach(() => {
    useSelector.mockReturnValue({ userInfo });
    wrapper = shallow(<Job />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the EditJobForm component when jobEditor is true', () => {
    //wrapper.setState({ jobEditor: true });
    wrapper = shallow(<Job />,{ jobEditor: true });
    expect(wrapper.find(EditJobForm)).toHaveLength(0);
  });

  it('should render the Jobs component when jobEditor is false', () => {
    //wrapper.setState({ jobEditor: false });
    wrapper = shallow(<Job />,{ jobEditor: false });
    expect(wrapper.find(Jobs)).toHaveLength(1);
  });

  it('should call setJobEditor with true when jobEditorMode is called', () => {   
    expect(wrapper.find(EditJobForm)).toHaveLength(0);
  });

  it('should call setJobEditor with false when quitJobEditor is called', () => {
    expect(wrapper.jobEditor).not.toBe(null);
  });

  it('should pass the correct props to Jobs component', () => {
    const jobsComponent = wrapper.find(Jobs);
    expect(jobsComponent).not.toBe(null);
  });

  it('should pass the correct props to EditJobForm component', () => {
    //wrapper.setState({ jobEditor: true, job: {} });
    //wrapper = shallow(<StoryApp />);
    wrapper = shallow(<Job />,{ jobEditor: true, job: {} });
    const editJobFormComponent = wrapper.find(EditJobForm);
    //console.log(editJobFormComponent);
    expect(editJobFormComponent).not.toBe(null);
  });
});
