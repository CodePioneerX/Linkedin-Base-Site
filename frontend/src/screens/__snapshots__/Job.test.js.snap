// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Component should render correctly 1`] = `ShallowWrapper {}`;
